src/compiler/checker.ts

```diff
@@ -17006,32 +17006,24 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         return type;
 
-        function addSpans(texts: readonly string[] | string, types: readonly Type[]): boolean {
-            const isTextsArray = isArray(texts);
+        function addSpans(texts: readonly string[], types: readonly Type[]): boolean {
             for (let i = 0; i < types.length; i++) {
                 const t = types[i];
-                const addText = isTextsArray ? texts[i + 1] : texts;
                 if (t.flags & (TypeFlags.Literal | TypeFlags.Null | TypeFlags.Undefined)) {
                     text += getTemplateStringForType(t) || "";
-                    text += addText;
-                    if (!isTextsArray) return true;
+                    text += texts[i + 1];
                 }
                 else if (t.flags & TypeFlags.TemplateLiteral) {
                     text += (t as TemplateLiteralType).texts[0];
                     if (!addSpans((t as TemplateLiteralType).texts, (t as TemplateLiteralType).types)) return false;
-                    text += addText;
-                    if (!isTextsArray) return true;
+                    text += texts[i + 1];
                 }
-                else if (isGenericIndexType(t) || isPatternLiteralPlaceholderType(t)) {
+                else if (isGenericIndexType(t) || isPatternLiteralPlaceholderType(t) || t.flags & TypeFlags.Intersection) {

```

Comment:
Actually, on second thought, you should update `isPatternLiteralPlaceholderType` itself rather than this condition. That'll handle the string mapping types a bit more consistently without needing your other change to the function, and will fix nesting these inside one another. Moreover, you probably want to say only intersections which contain at least one valid pattern literal placeholder type. I don't really think `a${{x: number} & {y: string}}` should work.

Reply:
Thanks for the tip, I'll take a look.

Reply:
To clarify, what "other change to the function" are you referring to in the above? Just `|| t.flags & TypeFlags.Intersection`?

I have tried:

```ts
        if (type.flags & TypeFlags.Intersection) {
            return some((type as IntersectionType).types, isPatternLiteralPlaceholderType);
        }
```

But, I run into the problem that `isPatternLiteralPlaceholderType` is false for string literals as those are always handled before getting to a call to `isPatternLiteralPlaceholderType`, so maybe I have to also check those even if they need special handling most of the time.

Reply:
I tried to make the change, but, maybe I went too far. Mainly, I had to make it handle literals too, and it seemed off to _not_ also allow other kinds of literals when I was already handling string literals.